### ISSUE TITLE
[v2] String fields disappear from schema when published with null values

### ISSUE THREAD

**Issue by jamesisaac** [datocms/gatsby-source-datocms#25]
I have a single-instance model (`Homepage`) with a non-required string field (`subtitle`).  If updated content gets published with no value in that field, the `gatsby develop` process breaks, as it starts reporting GraphQL errors:

```
success createPages — 0.000 s
success createPagesStatefully — 0.062 s
success onPreExtractQueries — 0.008 s
success update schema — 0.264 s
GraphQL Error Unknown field `subtitle` on type `DatoCmsHomepage`

  file: src/pages/index.js

   1 |
   2 |   query PageHomeQuery {
   3 |     homepage: datoCmsHomepage {
   4 |       title
>  5 |       subtitle
     |       ^
   6 |       seoMetaTags {
   7 |         ...GatsbyDatoCmsSeoMetaTags
   8 |       }
   9 |     }
  10 |   }
  11 |

success extract queries from components — 0.158 s
```

This way, a content editor can break a Gatsby site by just making a field blank (which should be a perfectly valid thing to do).

On DatoCMS's GraphQL playground, the field remains in the schema, just returning a null value, which makes a lot more sense.

I've tried digging into why this happens but not really familiar enough with the Gatsby plugin architecture to tell where the field is being dropped.   The closest I got is [here](https://github.com/datocms/gatsby-source-datocms/blob/0086ed54f37cf1c5992802283ef37b8d87a0401d/src/createItemNodes.js#L93), adding `if (fieldType === 'string' && itemNode[key] === null) { itemNode[key] = '' }` will cause the field to remain in the schema, so no errors, but obviously it's not a clean solution.

**Comment by jamesisaac** [comment]
Ah I see, thanks for helping narrow that down.  That does indeed seem like quite an oversight on Gatsby's part.

Nice workaround idea too.  For now I've just been using [patch-package](https://www.npmjs.com/package/patch-package) on this library to apply the change I mentioned above, which seems to work ok.

**Comment by shansmith01** [comment]
I like your work around. Which line did you insert your code at?

**Comment by stefanoverna** [comment]
Solved in version `2.1.0-beta.1`, soon to be released! 

https://github.com/datocms/gatsby-source-datocms/blob/2.1.0/CHANGELOG.md#v210
